src/features/game/expansion/components/UpgradeBuildingModal.tsx

```diff
@@ -73,12 +76,49 @@ export const UpgradeBuildingModal: React.FC<Props> = ({
     });
   };
 
+  const currentSupportedPlots = getSupportedPlots({
+    wellLevel: currentLevel,
+    buildings: state.buildings,
+  });
+
+  const nextSupportedPlots = getSupportedPlots({
+    wellLevel: nextLevel,
+    buildings: state.buildings,
+  });
+
+  const nextLevelFertility = nextSupportedPlots - currentSupportedPlots;
+
+  const buildingIcon =
+    buildingName === "Hen House"
+      ? HEN_HOUSE_VARIANTS[state.season.season][nextLevel]
+      : buildingName === "Water Well"
+        ? WATER_WELL_VARIANTS[state.season.season][nextLevel]
+        : BARN_IMAGES[state.island.type][state.season.season][nextLevel];
```

Comment:
I would suggest turning this into a function for simplicity/readability. It is recommended to avoid ternary's if:

- If it contains multiple conditions or expressions
- If the ternary extends across multiple lines and becomes difficult to read

```suggestion
function getBuildingIcon(buildingName: string, state: State, nextLevel: number) {
  const { season, island } = state;

  if (buildingName === "Hen House") {
    return HEN_HOUSE_VARIANTS[season.season][nextLevel];
  }

  if (buildingName === "Water Well") {
    return WATER_WELL_VARIANTS[season.season][nextLevel];
  }

  return BARN_IMAGES[island.type][season.season][nextLevel];
}
```